go/toolchain/toolchains.bzl

```diff
@@ -166,20 +129,26 @@ def declare_toolchains():
   # Use the final dictionaries to create all the toolchains
   for toolchain in _toolchains:
     if "default" not in toolchain:
+      goos, _, goarch = toolchain["target"].partition("_")
+      target_constraints = [
+          "@io_bazel_rules_go//go/toolchain:" + goos,
+          "@io_bazel_rules_go//go/toolchain:" + goarch,
+      ]
+      exec_constraints = ["@io_bazel_rules_go//go/toolchain:" + part for part in toolchain["host"].split("_")]
```

Comment:
Is there a difference in data format between `toolchain["target"]` and `toolchain["host"]`? They seem the same, but the code to build `exec_constraints` is different and seems to imply more constraints may be allowed. I guess it's just because `goos` and `goarch` are used below, but it might be good to build these lists the same way for readability.

Reply:
I originally wrote them both using the split form, but then I needed to cache the goos and goarch so I rewrote the first one.